transcom/mymove
# [MB-14989] Improve error handling for editing/adding ppm shipments as an office user

## [Jira ticket](https://dp3.atlassian.net/browse/MB-14989) for this change

## Summary
In this PR, I update the handling of server errors that occur when the first page of the add/edit shipment form is submitted (I did not look at the submission of the Advances page). 

The `ShipmentForm` component has a lot of conditional logic in it to change the form behavior slightly for different kinds of shipments and users. Server errors were not being handled consistently across all of the branches in the code. These are errors that are not the user's fault, and may be resolved if they simply resubmit the form.

Here are some of the different ways a failure to submit the closeout office could present depending on user/shipment type:
- user is redirected to Move Details page with an error message displayed
- user is given no indication that something went wrong, and the closeout office update is silently ignored
- user gets stuck on first page of `ShipmentForm` with no error message, and the submit button disabled

I updated all of the different branches to stay on the page where the error occurred and display an error message, so the user would have the option to resubmit the form without losing their work, or click cancel to go back to the Move Details page.

## Setup to Run Your Code

Start the UI locally.

```sh
make client_run
```
Start the Go server locally.

```sh
make server_run
```




### Additional steps
There are two network requests that can fail, the request to update the closeout office and the request to update the shipment. You can block one of these at a time in your browser dev tools, and then run through the following scenarios.
**In every scenario, the behavior you want to see is: when you try to submit the first page of the shipment form, you should stay on the first page and see an error message.**

This screen recording shows how to block a network request with Chrome's devtools (firefox is really similar), and demonstrates editing a PPM.
https://user-images.githubusercontent.com/2627844/219819996-9bdd29d8-8401-4569-8121-5c367199c6eb.mov

This screen recording demonstrates testing adding a PPM shipment with the closeout office update request blocked:
https://user-images.githubusercontent.com/2627844/219820715-234ee235-2994-4501-abae-af861d4c9ec1.mov

### Testing steps

- First create or find an Army or Air Force move that a counselor can add or edit shipments on. (eg **PPMSC1** in devseed data). You should be able to use this same move for all test cases

Test with `DELETE /ghc/v1/moves/*/closeout-office` blocked:
- edit PPM as Service Counselor
- create PPM as Service Counselor
- edit PPM shipment as TOO

Test with `PATCH /ghc/v1/move_task_orders/*/mto_shipments/*` blocked:
- create non-PPM shipment as SC
- edit non-PPM shipment as SC
- edit non-PPM shipment as TOO



## Verification Steps for Author


## Verification Steps for Reviewers

These are to be checked by a reviewer.



### Frontend
